### DISCUSSION TITLE
How about python support?

### DISCUSSION THREAD

**Discussion by Bloodielie** [uNetworking/uWebSockets#1202] [Q&A] [answered]
I saw that you did Python support, but for some reason deleted all the files. Why did you do this, and should we expect python support in the future?

**Comment by ghost** [comment, marked as answer]
I made the python repo private. It is not deleted only hidden. The problem was that nobody seemed to care, and Python has issues with fragmentation such as PyPy being much faster than CPython but lacking C extensions. Also the ASGI thing required extensive performance loss for compatibility since it requires Futures from the asyncio library, which are really, really slow. I will probably not try to get into the Python market since those people are a whole different group and don't seem to know anything about uWS whatsoever. It would be like starting from square 1 and I don't want that.

**Reply by Bloodielie** [reply]
It's a pity you hid the repository. I'm really interested in testing your library in python.

**Reply by ghost** [reply]
I can make it public again but I can't support it

**Reply by Bloodielie** [reply]
I would be glad if you open the repository even without code support.

**Reply by AmirHmZz** [reply]
@alexhultman Hello! I've just met your graceful and fascinating super fast library! You've made a great job here and why you are hiding it from python community!? I know the head developers of python core are really unfriendly, But I believe python community are people like me and @Bloodielie who are looking for these great jobs. I think python really needs uWS :(

**Reply by AmirHmZz** [reply]
@alexhultman Any way I want to develop an extension for python language, but I just need some starting points to get it working...

**Reply by ghost** [reply]
https://github.com/uNetworking/uWebSockets/pull/1395

**Reply by cirospaciari** [reply]
@Bloodielie here its my repo with some python support, https://github.com/cirospaciari/socketify.py i plan to support all C API from PR #1395 and expand the C API to HTTP3 support and more needed functions. I plan to fully support Python3, PyPy3, Ruby 3.1+ and Luajit. Here is the work in progress in Ruby: https://github.com/cirospaciari/socketify.rb. I will only support Lua after Python/PyPy and Ruby, Join the Discussion in https://github.com/cirospaciari/socketify.py/discussions if you want to try, ask something or give some feedback
